### PR TITLE
TicketValidationException<->UnrecognizableServiceForServiceTicketValidationException

### PR DESCRIPTION
**Pull request by mmoayyed**
Attempts to handle #561 

A TicketValidationException is only thrown if a service != service ticket. I have made the exception more generic and derived a specialized case that deals more with INVALID_SERVICE.

### CODE REVIEW

File: cas-server-core/src/main/java/org/jasig/cas/ticket/TicketValidationException.java [hunk 1]
```diff
@@ -24,7 +24,6 @@
  * Exception to alert that there was an error validating the ticket.
  *
  * @author Scott Battaglia
-
  * @since 3.0.0
  */
 public class TicketValidationException extends TicketException {
```
Comment (leleuj) [inline cas-server-core/src/main/java/org/jasig/cas/ticket/TicketValidationException.java]: Don't you want to make it `abstract` as the root hierarchy of the ticket validation exception?

File: cas-server-core/src/main/java/org/jasig/cas/ticket/UnrecognizableServiceForServiceTicketValidationException.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import org.jasig.cas.authentication.principal.Service;
+
+/**
+ * An exception that may be thrown during service ticket validation
+ * to indicate that the service ticket is not valid and was not originally
+ * issued for the submitted service.
+ * @author Misagh Moayyed
+ * @since 4.1
+ */
+public class UnrecognizableServiceForServiceTicketValidationException extends TicketValidationException {
```
Comment (leleuj) [inline cas-server-core/src/main/java/org/jasig/cas/ticket/UnrecognizableServiceForServiceTicketValidationException.java]: I feel the name of the exception to be a bit too long, but I don't really have a better proposal.

**Comment by leleuj** [comment]
+1 with two comments.

**Comment by mmoayyed** [comment]
I updated the pull to mark the class as abstract. It is also now marked as deprecated because it correctly should renamed AbstractTicketValidation later on.

**Pull request merged**